distributed/utils.py

```diff
@@ -898,6 +899,22 @@ def nbytes(frame):
             return frame.nbytes
 
 
+class ThrottledGC(object):
+    """
+    Allows to run throttled garbage collection as a countermeasure to e.g.:
+    https://github.com/dask/zict/issues/19
+    """
+    def __init__(self, min_interval_in_sec=1.0):
+        self.min_interval_in_sec = min_interval_in_sec
+        self.last_collect = time()
+        #gc.collect()
+
+    def collect(self, force_gc=False):
+        new_time = time()
+        if force_gc or new_time - self.last_collect > self.min_interval_in_sec:
+            gc.collect()
+            self.last_collect = new_time
```

Comment:
As @pitrou said in another PR, GC can legitimately take more than several hundred milliseconds (maybe even seconds) on interpreters loaded with lots of small python objects (e.g. in dask.bags with nested constructs). Therefore you might want to consider a strategy that records the time taken by the last call to `gc.collect()` and the time elapsed since after the end to last call to `gc.collect()` so as to only call `gc.collect()` if less than 5% of the time is spent inside the `gc.collect()` calls in aggregate instead of using a fixed absolute value for `min_interval_in_sec`.

Also using a [time.monotonic()](https://docs.python.org/3/library/time.html#time.monotonic) instead of `time.time()` under Python 3.5 and later will probably spare some rare but potentially hard to debug issues in case of leap second events.

Reply:
Yes, good point, I think I'll just use your implementation, no need for two throttled GC implementations ;). I'll still have to check if this PR is needed at all with you modification.